app/controllers/krikri/providers_controller.rb

```diff
@@ -0,0 +1,19 @@
+module Krikri
+  # This models a provider that is stored in Marmotta.
+  class ProvidersController < ApplicationController
+    def index
+      @providers = Krikri::Provider.all
+    end
+
+    def show
+      if params[:set_session]
+        session[:current_provider] = params[:id]
+        redirect_to :back, provider: params[:id]
+      elsif params[:clear_session]
+        session.delete :current_provider
+        redirect_to providers_path
```

Comment:
What we're really shooting for is something like

```pseudocode
  redirect_to "#{request.referrer}?provider=#{params[:id]}"
```
 unless `request.referrer` doesn't accept a `provider`.  @AudreyAltman made the strong point that this really needs to be a `POST` request, and i'm of the opinion that `session` should have its own controller.  The POST + redirect could be handled, ideally, in javascript, with only the navbar reloaded if the redirect turns out to be unnecessary.

Anyway, I'm increasingly of the opinion that this should wait.  We can punt it to a story like: 

> As a QA user, I want to be able to select a provider (or "All Providers") from the nav bar, and get the current content filtered to provider.